### PR TITLE
Add feed fixen

### PR DESCRIPTION
**Pull request by urfolomeus**
Through a previous feature we had accidentally created a rare situation where, if you had no source rules, it was possible to get stuck on the apply rules page. This PR fixes that issue.

### CODE REVIEW

File: app/models/feed.rb [hunk 1]
```diff
@@ -16,11 +16,12 @@ def applicable_source_rules
   end
 
   def update_applied_rules(selected_rule_ids)
+    return false if selected_rule_ids.nil?
+
     remove_rules missing_rule_ids(selected_rule_ids)
```
Comment (Tr4pSt3R) [inline app/models/feed.rb]: there's no test to clarify the definition of a 'missing rule' and to test that they can be successfully removed if they are indeed missing. 
Apart from this all else looks spotless

Comment (urfolomeus) [inline app/models/feed.rb]: Thanks @Tr4pSt3R. The intention of the context starting on line 199 is to test this. Is it not enough?

**Comment by neoeno** [comment]
:+1:

File: app/models/feed.rb (hunk 1 above)
Comment (Tr4pSt3R) [inline app/models/feed.rb]: ```context 'feed has applied rule, not in array' do``` yup that looks fine. What is array? I know it's not part of your changes but is there a better name for it please?
Everything else is fine
:+1:

Comment (urfolomeus) [inline app/models/feed.rb]: Sounds good. Will fix just now, thanks.

Comment (urfolomeus) [inline app/models/feed.rb]: Done @Tr4pSt3R.

**Comment by Tr4pSt3R** [comment]
@urfolomeus :tada: it's a ton more easier to understand. Thanks :+1: :ship:

**Comment by urfolomeus** [comment]
Thanks for all your feedback, @Tr4pSt3R. Appreciated :)

**Pull request merged**